File: openmp-wg-pII-2015/omp_condition.c
#include <omp.h>
#include <stdio.h>

int main(int argc,char *argv[]){
    int N = 10;
    if(argc > 1)
        N = atoi(argv[1]);
    int i = 0;
    #pragma omp parallel for if(N>10)
    for(i = 0;i < N;i++)
        printf("Thread %d gets i = %d\n",omp_get_thread_num(),i);
} 

File: openmp-wg-pII-2016/omp_condition.f90
program omp_condition

    use omp_lib

    character(10) :: input
    integer :: i,input_val
! read the command line argument
    call getarg(1,input)
!convert to integer
    read(input,*) input_val
!$OMP PARALLEL DO if(input_val>10)
    do i=1,input_val
        write(*,"(A,I4,A,I4)") "Thread ",omp_get_thread_num()," gets i = ",i
    enddo
!$OMP END PARALLEL DO

end program
